File: src/main/java/io/golos/commun4J/model/Beneficiary.java
package io.golos.commun4J.model;

import com.memtrip.eos.abi.writer.Abi;
import com.memtrip.eos.abi.writer.IntCompress;
import com.memtrip.eos.abi.writer.NameCompress;

@Abi
public class Beneficiary {
    //deductprcnt
    private String account;
    private int deductprcnt;

    public Beneficiary(CommunName account, int deductprcnt) {
        this.account = account.getName();
        this.deductprcnt = deductprcnt;
    }

    @NameCompress
    public String getAccount() {
        return account;
    }

    public void setAccount(String account) {
        this.account = account;
    }

    @IntCompress
    public int getDeductprcnt() {
        return deductprcnt;
    }

    public void setDeductprcnt(int deductprcnt) {
        this.deductprcnt = deductprcnt;
    }
}

File: src/main/java/io/golos/commun4J/model/Beneficiary.kt
package io.golos.commun4J.model

import com.memtrip.eos.abi.writer.Abi
import com.memtrip.eos.abi.writer.IntCompress
import com.memtrip.eos.abi.writer.NameCompress

@Abi
class Beneficiary(private val account: CommunName,
                  private val deductprcnt: Int) {


    val getAccount: String
        @NameCompress get() = account.name

    val getDeduct: Int
        @IntCompress get() = deductprcnt

}
